interop/xds/client/client.go

```diff
@@ -95,25 +124,90 @@ func (s *statsService) GetClientStats(ctx context.Context, in *testpb.LoadBalanc
 	// Wait until the requested RPCs have all been recorded or timeout occurs.
 	for {
 		select {
-		case r := <-watcher.c:
-			if r != nil {
-				watcher.rpcsByPeer[(*r).GetHostname()]++
+		case info := <-watcher.chanHosts:
+			if info != nil {
+				watcher.rpcsByPeer[info.hostname]++
+
+				rpcsByPeerForType := watcher.rpcsByType[info.typ]
+				if rpcsByPeerForType == nil {
+					rpcsByPeerForType = make(map[string]int32)
+					watcher.rpcsByType[info.typ] = rpcsByPeerForType
+				}
+				rpcsByPeerForType[info.hostname]++
 			} else {
 				watcher.numFailures++
 			}
 			watcher.remainingRpcs--
 			if watcher.remainingRpcs == 0 {
-				return &testpb.LoadBalancerStatsResponse{NumFailures: watcher.numFailures + watcher.remainingRpcs, RpcsByPeer: watcher.rpcsByPeer}, nil
+				return watcher.buildResp(), nil
 			}
 		case <-ctx.Done():
 			grpclog.Info("Timed out, returning partial stats")
-			return &testpb.LoadBalancerStatsResponse{NumFailures: watcher.numFailures + watcher.remainingRpcs, RpcsByPeer: watcher.rpcsByPeer}, nil
+			return watcher.buildResp(), nil
+		}
+	}
+}
+
+type rpcType string
+
+const (
+	unaryCall rpcType = "UnaryCall"
+	emptyCall rpcType = "EmptyCall"
+)
+
+func parseRPCTypes(rpcStr string) (ret []rpcType) {
+	rpcs := strings.Split(rpcStr, ",")
+	if len(rpcs) == 0 {
```

Comment:
Optional: `if len(rpcStr) == 0` and move above the `Split`.

Reply:
Done